server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -1058,6 +1081,26 @@ private DefaultSearchContext createSearchContext(ReaderContext reader, ShardSear
         return searchContext;
     }
 
+    private boolean shouldReverseLeafReaderContexts(ShardSearchRequest request) {
+        // By default, if search query is on desc order, we will search leaves in reverse order.
+        // This is beneficial for desc order sort queries on time series based workload,
+        // where recent data is always on new segments which are in last.
+        // This won't regress or impact other type of workload where data is randomly distributed
+        // across segments. So turning it on by default.
+        // searchSegmentOrderReversed is true by default
+        if (searchSegmentOrderReversed) {
+            // Only reverse order for desc order sort queries
+            if (request != null
+                && request.source() != null
+                && request.source().sorts() != null
+                && request.source().sorts().size() > 0
+                && request.source().sorts().get(0).order() == SortOrder.DESC) {
+                return true;
+            }
+        }
+        return false;
+    }
```

Comment:
Could we cover this logic in a UT in SearchServiceTests?

Reply:
This would be too small to cover in UT. I think this will very well covered in integ tests since this boolean setting is enabled by default.